### ISSUE TITLE
Error occured during push; Stack UPDATE_ROLLBACK_FAILED state and can not be updated. 

### ISSUE THREAD

**Issue by jncheung** [aws-amplify/amplify-cli#6480]
**Describe the bug**

Error during `amplify function push` and cannot rollback in the Cloudformation console

**To Reproduce**

I developed an external lambda layer using the cdk and try import the layer back for the lambda functions using `amplify update functions` ,

Since the layer involves the use of Appsync, and therefore I also granted the `api` access via `amplify update function`  > Pick my function > Resource access permissions > api

![image](https://user-images.githubusercontent.com/30644696/105827103-899b8480-5ffc-11eb-9365-644790d4830a.png)

Since only the `function` category was involved in this update and so i ran `amplify function push`, and return this error: 
```
Circular dependency between resources: [UpdateRolesWithIDPFunctionOutputs, apiDesklessWorkerGraphQL, functioncreateOrganisationMakeCreaterAsOwner, authDesklessWorkers, functionupdateSendbirdUserSession, UpdateRolesWithIDPFunction, functionDesklessWorkersPostConfirmation]
```

![image](https://user-images.githubusercontent.com/30644696/105827377-e0a15980-5ffc-11eb-930a-a057d5bc7286.png)

Well, since the cli said the Circular dependency thing, then I removed the api access permission from my `PostConfirmation` trigger.

I also made few changes on the schemas about the `@aws_iam` and
`@aws_cognito_user_pools` for my custom mutations and a new version for my Lambda layer deployed via cdk.

so I ran `amplify update function` again to update the layer version like this:

![image](https://user-images.githubusercontent.com/30644696/105828744-74bff080-5ffe-11eb-98b2-ef0c3470f17f.png)

However, I decided to **NOT** update those Graphql changes in this push and run `amplify function push` again and update my lambda functions, and return two errors: 

![image](https://user-images.githubusercontent.com/30644696/105828991-c4062100-5ffe-11eb-9f0c-a811531648bf.png)

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system
4.41.1

**Expected behavior**

**Screenshots**
Pasted above
**Desktop (please complete the following information):**

- OS: [e.g. Mac/Windows/Ubuntu]
`12.19.0`

**Log output**
Include any relevant log output under `~/.amplify/logs/amplify-cli-<issue-date>.log`

```
2021-01-26T07:45:36.087Z|error : amplify-provider-awscloudformation.push-resources.updateCloudFormationNestedStack(["/Users/jason.cheung/workspace/DesklessWorkerTesting/amplify/backend/awscloudformation","/Users/jason.cheung/workspace/DesklessWorkerTesting/amplify/backend/awscloudformation/nested-cloudformation-stack.yml"])
ValidationError: Circular dependency between resources: [UpdateRolesWithIDPFunctionOutputs, apiDesklessWorkerGraphQL, functioncreateOrganisationMakeCreaterAsOwner, authDesklessWorkers, functionupdateSendbirdUserSession, UpdateRolesWithIDPFunction, functionDesklessWorkersPostConfirmation]
2021-01-26T07:45:36.100Z|error : amplify-provider-awscloudformation.push-resources.run([{"resourcesToBeCreated":[],"resourcesToBeUpdated":[{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]orkersPostConfirmation-[***]ure","Name":"[***]tConfirmation-[***]ure","LambdaExecutionRole":"[***]tConfirmation-feature"},"lastPushTimeStamp":"2021-01-26T07:42:14.695Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/[***]orkersPostConfirmation-cloudformation-template.json","logicalId":"[***]PostConfirmation"},"lastBuildTimeStamp":"2021-01-26T07:29:11.641Z","lastPackageTimeStamp":"2021-01-26T07:29:12.502Z","distZipFilename":"[***]tConfirmation-[***]624a644f-[***].zip","lastPushDirHash":"xg50TVFwsxWoc2EYwCvywyZxWS0=","dependsOn":[{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"resourceName":"[***]tConfirmation","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","dependsOn":[{"category":"auth","resourceName":"[***]orkers","attributes":["UserPoolId"]},{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"lastBuildTimeStamp":"2021-01-26T07:08:24.853Z","lastPackageTimeStamp":"2021-01-26T07:08:26.759Z","distZipFilename":"[***]eCreaterAsOwner-[***]6e376448-[***].zip","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/[***]eCreaterAsOwner-cloudformation-template.json","logicalId":"[***]MakeCreaterAsOwner"},"lastPushTimeStamp":"2021-01-26T07:42:48.773Z","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]sationMakeCreaterAsOwner-[***]ure","Name":"[***]eCreaterAsOwner-[***]ure","LambdaExecutionRole":"desklessworkersLambdaRole7886e729-feature"},"lastPushDirHash":"5DMBeY/OwUy3OsiZh0T8DFqEhI0=","resourceName":"[***]eCreaterAsOwner","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","dependsOn":[{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"lastBuildTimeStamp":"2021-01-26T07:08:24.855Z","lastPackageTimeStamp":"2021-01-26T07:08:25.053Z","distZipFilename":"[***]serSession-[***]75755248-[***].zip","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/[***]serSession-cloudformation-template.json","logicalId":"[***]irdUserSession"},"lastPushTimeStamp":"2021-01-26T07:42:36.277Z","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]eSendbirdUserSession-[***]ure","Name":"[***]serSession-[***]ure","LambdaExecutionRole":"desklessworkersLambdaRole70d717bf-feature"},"lastPushDirHash":"dhhXfmcS9wpHhzxPNjt7CdNOIPc=","resourceName":"[***]serSession","category":"function"}],"resourcesToBeSynced":[],"resourcesToBeDeleted":[],"tagsUpdated":false,"allResources":[{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]rkersDefineAuthChallenge-[***]ure","Name":"[***]eAuthChallenge-[***]ure","LambdaExecutionRole":"[***]eAuthChallenge-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.281Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersDefineAuthChallenge-cloudformation-template.json","logicalId":"[***]fineAuthChallenge"},"dependsOn":[],"lastBuildTimeStamp":"2021-01-26T07:08:24.848Z","lastPackageTimeStamp":"2021-01-26T07:08:25.041Z","distZipFilename":"[***]eAuthChallenge-[***]30427873-[***].zip","lastPushDirHash":"5Faxa1BoznzevMli+sl3q2e0WvY=","resourceName":"DesklessWorkersDefineAuthChallenge","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]rkersCreateAuthChallenge-[***]ure","Name":"[***]eAuthChallenge-[***]ure","LambdaExecutionRole":"[***]eAuthChallenge-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.289Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersCreateAuthChallenge-cloudformation-template.json","logicalId":"[***]eateAuthChallenge"},"resourceName":"DesklessWorkersCreateAuthChallenge","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]VerifyAuthChallengeResponse-[***]ure","Name":"[***]ChallengeResponse-[***]ure","LambdaExecutionRole":"[***]ChallengeResponse-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.291Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersVerifyAuthChallengeResponse-cloudformation-template.json","logicalId":"[***]uthChallengeResponse"},"resourceName":"DesklessWorkersVerifyAuthChallengeResponse","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]rkersPostAuthentication-[***]ure","Name":"[***]Authentication-[***]ure","LambdaExecutionRole":"[***]Authentication-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.294Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersPostAuthentication-cloudformation-template.json","logicalId":"[***]ostAuthentication"},"resourceName":"DesklessWorkersPostAuthentication","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]orkersPostConfirmation-[***]ure","Name":"[***]tConfirmation-[***]ure","LambdaExecutionRole":"DesklessWorkersPostConfirmation-feature"},"lastPushTimeStamp":"2021-01-26T07:42:14.695Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersPostConfirmation-cloudformation-template.json","logicalId":"[***]PostConfirmation"},"lastBuildTimeStamp":"2021-01-26T07:29:11.641Z","lastPackageTimeStamp":"2021-01-26T07:29:12.502Z","distZipFilename":"[***]tConfirmation-[***]624a644f-[***].zip","lastPushDirHash":"xg50TVFwsxWoc2EYwCvywyZxWS0=","dependsOn":[{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"resourceName":"DesklessWorkersPostConfirmation","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]orkersPreAuthentication-[***]ure","Name":"[***]uthentication-[***]ure","LambdaExecutionRole":"[***]uthentication-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.297Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersPreAuthentication-cloudformation-template.json","logicalId":"[***]reAuthentication"},"resourceName":"DesklessWorkersPreAuthentication","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]lessWorkersPreSignup-[***]ure","Name":"[***]sPreSignup-[***]ure","LambdaExecutionRole":"[***]sPreSignup-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.299Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersPreSignup-cloudformation-template.json","logicalId":"[***]kersPreSignup"},"resourceName":"DesklessWorkersPreSignup","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]rkersPreTokenGeneration-[***]ure","Name":"[***]okenGeneration-[***]ure","LambdaExecutionRole":"[***]okenGeneration-feature"},"lastPushTimeStamp":"2021-01-26T07:31:09.301Z","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/DesklessWorkersPreTokenGeneration-cloudformation-template.json","logicalId":"[***]reTokenGeneration"},"resourceName":"DesklessWorkersPreTokenGeneration","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","dependsOn":[{"category":"auth","resourceName":"DesklessWorkers","attributes":["UserPoolId"]},{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"lastBuildTimeStamp":"2021-01-26T07:08:24.853Z","lastPackageTimeStamp":"2021-01-26T07:08:26.759Z","distZipFilename":"[***]eCreaterAsOwner-[***]6e376448-[***].zip","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/createOrganisationMakeCreaterAsOwner-cloudformation-template.json","logicalId":"[***]MakeCreaterAsOwner"},"lastPushTimeStamp":"2021-01-26T07:42:48.773Z","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]sationMakeCreaterAsOwner-[***]ure","Name":"[***]eCreaterAsOwner-[***]ure","LambdaExecutionRole":"desklessworkersLambdaRole7886e729-feature"},"lastPushDirHash":"5DMBeY/OwUy3OsiZh0T8DFqEhI0=","resourceName":"createOrganisationMakeCreaterAsOwner","category":"function"},{"build":true,"providerPlugin":"[***]rmation","service":"Lambda","dependsOn":[{"category":"api","resourceName":"[***]erGraphQL","attributes":["GraphQLAPIIdOutput","GraphQLAPIEndpointOutput"]}],"lastBuildTimeStamp":"2021-01-26T07:08:24.855Z","lastPackageTimeStamp":"2021-01-26T07:08:25.053Z","distZipFilename":"[***]serSession-[***]75755248-[***].zip","providerMetadata":{"s3TemplateURL":"https://s3.amazonaws.com/amplify-desklessworkers-feature-155419-deployment/amplify-cfn-templates/function/updateSendbirdUserSession-cloudformation-template.json","logicalId":"[***]irdUserSession"},"lastPushTimeStamp":"2021-01-26T07:42:36.277Z","output":{"Region":"us-east-2","Arn":"[***]mbda:us-[***]st-[***]eSendbirdUserSession-[***]ure","Name":"[***]serSession-[***]ure","LambdaExecutionRole":"desklessworkersLambdaRole70d717bf-feature"},"lastPushDirHash":"dhhXfmcS9wpHhzxPNjt7CdNOIPc=","resourceName":"updateSendbirdUserSession","category":"function"}]}])
ValidationError: Circular dependency between resources: [UpdateRolesWithIDPFunctionOutputs, apiDesklessWorkerGraphQL, functioncreateOrganisationMakeCreaterAsOwner, authDesklessWorkers, functionupdateSendbirdUserSession, UpdateRolesWithIDPFunction, functionDesklessWorkersPostConfirmation]
```
**Additional context**
I have tried to do the `continue update rollback` trick in the cloudformation console but no good result.

![image](https://user-images.githubusercontent.com/30644696/105830046-fc5a2f00-5fff-11eb-8f5a-ec59ec8f30db.png)

**Comment by ammarkarachi** [comment]
@jncheung It looks like it's still looking for earlier lambda layer version from the stack trace. Can you check if your profile has access to the lambda layer?

**Comment by jncheung** [comment]
@ammarkarachi I have the AWS managed policy  `arn:aws:iam::aws:policy/AWSLambda_FullAccess` in my profile IAM setting.
I should have all access to lambda related services.

I also added the GetLayerVersion policy to my PostConfirmation Trigger for layer version 15 and 16, and tried the continue rollback but no good result

![image](https://user-images.githubusercontent.com/30644696/105929353-5996c480-6082-11eb-90b8-7975bc84ee3c.png)
![image](https://user-images.githubusercontent.com/30644696/105929436-85b24580-6082-11eb-8e82-a0f0081d3516.png)

**Comment by jncheung** [comment]
@ammarkarachi maybe we can do a rollback to a previous deployment. but i cannot find such option. Any recommendations?

**Comment by jncheung** [comment]
I found out that this is a problem related to CDK `layerVersion` L2 construct which did not create a new version.

Thanks for your help @ammarkarachi, you gave me a hint of where I should dig into.

I will close the issue now but the rollback feature is actually a good to have. 

If someone ran into this situation, we still hoping there is some chances to rollback, In my case, I moved my latest version to a new amplify environment and deleted the old one and which I will have to move the all data to the new env as well.
